mmdet/models/losses/iou_loss.py

```diff
@@ -88,28 +88,9 @@ def giou_loss(pred, target, eps=1e-7):
     Return:
         Tensor: Loss tensor.
     """
-    # overlap
-    lt = torch.max(pred[:, :2], target[:, :2])
-    rb = torch.min(pred[:, 2:], target[:, 2:])
-    wh = (rb - lt).clamp(min=0)
-    overlap = wh[:, 0] * wh[:, 1]
-
-    # union
-    ap = (pred[:, 2] - pred[:, 0]) * (pred[:, 3] - pred[:, 1])
-    ag = (target[:, 2] - target[:, 0]) * (target[:, 3] - target[:, 1])
-    union = ap + ag - overlap + eps
-
-    # IoU
-    ious = overlap / union
-
-    # enclose area
-    enclose_x1y1 = torch.min(pred[:, :2], target[:, :2])
-    enclose_x2y2 = torch.max(pred[:, 2:], target[:, 2:])
-    enclose_wh = (enclose_x2y2 - enclose_x1y1).clamp(min=0)
-    enclose_area = enclose_wh[:, 0] * enclose_wh[:, 1] + eps
-
-    # GIoU
-    gious = ious - (enclose_area - union) / enclose_area
+    gious = bbox_gious(pred, target, is_aligned=True)
+    if gious.ndim >= 2 and gious.size(-1) == 1:
```

Comment:
1. Is this check necessary? Since the squeeze operation is done in bbox_gious.
2. Please fully test this case in unit tests if they are necessary.